### DISCUSSION TITLE
No data comes to SWR variable

### DISCUSSION THREAD

**Discussion by vtarelkin** [vercel/swr#1338] [Q&A]
Please consider such component.

I sucessfully get image from server but I cannot see any data in console log. 

Also I can't access SWR's error - it tells me error is not defined. And if I move 

`    const {data, error} = useSWR(imageId && 'someUrl\getImage', getExistingFile);
`
above it tells me that getExistingFile function is not defined. 

Also I wonder how could I implement cache for my images in future - to avoid downloading them again and again

```
const Image = ({imageId}) => {

    const getExistingFile = (url) => {

        axios
            .get(
                url,
                {
                    responseType: 'blob'
                }
            )
            .then (response => response.data)
           //  .catch(error => error(error));  --- it tells me it cannot access it! 
    };

    const {data, error} = useSWR(imageId && 'someUrl\getImage', getExistingFile);

    console.log("data", data)  // here is undefined

    return (
      <>
      </>
    );
};

export default Image;

```

**Comment by shuding** [comment]
Your fetcher is missing `return`.

**Reply by vtarelkin** [reply]
> Your fetcher is missing `return`.

https://swr.vercel.app/docs/data-fetching

there is no return in axios section too

I'll try anyway, thanks.

What about error? How can I use it in my code?
And what about cache?
